src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -85,8 +86,12 @@ internal abstract partial class AbstractAsynchronousTaggerProvider<TTag> where T
         /// <summary>
         /// Comparer used to check if two tags are the same.  Used so that when new tags are produced, they can be
         /// appropriately 'diffed' to determine what changes to actually report in <see cref="ITagger{T}.TagsChanged"/>.
+        /// <para>
+        /// Subclasses should always override this.  It is only virtual for binary compat.
```

Comment:
@MariaSolOs  for visibility.  But we should also make sure we get TS off of having their own subclass of this type.

Reply:
Another case where we could use a new attribute `ShouldOverrideAttribute` that triggers warning reported by the compiler/analyzer.